src/include/sof/audio/asrc/asrc_farrow.h

```diff
@@ -0,0 +1,633 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2012-2019 Intel Corporation. All rights reserved.
+ */
+
+/*
+ * @brief    API header containing sample rate converter struct and interface
+ *           functions.
+ *
+ * @mainpage Hifi3 Implementation of the Intel TSD Sample Rate Converter
+ *
+ * The sample rate converter is based on the so-called Farrow structure.
+ * It supports multi-channel operation, and the PCM samples can be
+ * represented by int16_t or int32_t values.
+ *
+ * The sample rate converter can be applied for transmit and receive
+ * use cases.  To support both directions, the sample rate converter
+ * offers two modes of operation: push-mode and pull-mode.
+ * <p>
+ * @image html Push_mode-Pull_mode-Use_case_150dpi.png
+ * "Push-mode vs. pull-mode operation"
+ * @image rtf  Push_mode-Pull_mode-Use_case_150dpi.png
+ * </p>
+ * If the sample rate converter operates in push-mode, the caller can
+ * specify how many input frames shall be processed with each call of
+ * this function.  Depending on the current conversion ratio, the
+ * number of output frames might vary from call to call. Therefore,
+ * the push-mode sample rate converter is usually combined with a ring
+ * buffer at its output. The push-mode operation is performed by the
+ * functions process_push16() and process_push32().
+ *
+ * If the sample rate converter operates in pull-mode, the caller can
+ * specify how many output frames shall be generated with each call of
+ * this function.  Depending on the current conversion ratio, the
+ * number of input frames might vary from call to call. Therefore, the
+ * pull-mode sample rate converter is usually combined with a ring
+ * buffer at its input. The pull-mode operation is performed by the
+ * functions process_pull16() and process_pull32().
+ */
+
+#ifndef IAS_SRC_FARROW_H
+#define IAS_SRC_FARROW_H
+
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+
+/*
+ * @brief Define whether the input and output buffers shall be
+ * interleaved or not.
+ */
+enum asrc_io_format {
+	IOF_DEINTERLEAVED, /*<! Non-interleaved, i.e. individual buffers */
```

Comment:
I think all asrc-specific constants should be prefixed with ASRC_

Reply:
Yep, will do.